tests/check/check.go

```diff
@@ -0,0 +1,108 @@
+package check
+
+import (
+	"context"
+	"testing"
+
+	parser "github.com/craigpastro/openfga-dsl-parser/v2"
+	"github.com/openfga/openfga/assets"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/stretchr/testify/require"
+	pb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+	"gopkg.in/yaml.v3"
+)
+
+type checkTests struct {
```

Comment:
Everything from L17-L46 is common code between `oldcheck` and `check`. It'd be nice if we consolidated these under a common top-level package `check` and just had to different test entrypoints

```
// file: tests/check/check.go
package check

... // boilerplate

func RunSchema1_0_CheckTests(...) {...}

func RunSchema1_1_CheckTests(...) {...}
```

These two test bodies can share common code but with the two primary differences injected in. They appear to only differ in `typesystem.SchemaVersion1_1` vs `typesystem.SchemaVersion1_0` AND `parser/v2` vs `parser`.

Reply:
Done! Thanks.